cmd/wire/main.go

```diff
@@ -66,11 +67,16 @@ func main() {
 		err = generate(os.Args[1:]...)
 	default:
 		fmt.Fprintln(os.Stderr, usage)
-		os.Exit(64)
+		exitCode = 64
 	}
 	if err != nil {
 		fmt.Fprintln(os.Stderr, "wire:", err)
-		os.Exit(1)
+		if exitCode == 0 {
```

Comment:
This strikes me as an odd conditional. Can you help me understand why this check is needed?

Reply:
Because if `diff` ran into problems, it returned `2, err`, and I don't want to clobber the `2` here. I added a comment to clarify.